dsp/src/testing.rs

```diff
@@ -0,0 +1,54 @@
+use super::Complex;
+
+pub fn f32_is_close(a: f32, b: f32) -> bool {
+    (a - b).abs() <= a.abs().max(b.abs()) * f32::EPSILON
+}
+
+pub fn complex_is_close(a: Complex<f32>, b: Complex<f32>) -> bool {
+    f32_is_close(a.0, b.0) && f32_is_close(a.1, b.1)
+}
+
+pub fn complex_array_is_close(a: &[Complex<f32>], b: &[Complex<f32>]) -> bool {
+    let mut result: bool = true;
+    a.iter().zip(b.iter()).for_each(|(i, j)| {
+        result &= complex_is_close(*i, *j);
+    });
+    result
+}
+
+pub fn within_tolerance(
```

Comment:
Can we stay closer to existing namings, e.g. from numpy: `allclose()`, `rtol`, `atol` (not "fixed").